lib/src/cli/commands/command_mixin.dart

```diff
@@ -74,6 +74,10 @@ Affects only local references.
     } else {
       throw ArgumentError('Unknown package ref: ${ref.ref}');
     }
+    // merge sources to not copy children of a parent separately
+    // => remove all sources that have a parent in the list
+    sources.removeWhere((sToRemove) => sources.any(
+        (s) => s != sToRemove && p.isWithin(s.sourceDir, sToRemove.sourceDir)));
```

Comment:
according to documentation of `isWithin` the it returns false for two similar path so I guess the equality check is redundant 
```suggestion
        (s) => p.isWithin(s.sourceDir, sToRemove.sourceDir)));
```

Reply:
nice find. Thanks for the pointer